src/main/java/pl/edu/mimuw/chess/Board.java

```diff
@@ -0,0 +1,108 @@
+package pl.edu.mimuw.chess;
+
+import java.util.ArrayList;
+
+public class Board {
+    private Piece[][] pieces;
+    private Color whoseTurn;
+    private int movesSinceBeating;
+    public static Color oppositeColor(Color color) {
+        if(color == Color.white) return Color.black;
+        if(color == Color.black) return Color.white;
+        return Color.empty;
+    }
+    public Board() {
+        pieces = new Piece[8][8];
+        pieces[0][0] = new Rook(this, Color.white);
```

Comment:
każda linia kodu jest zduplikowana dla white/black - to można wyciągnąc do funkcji pomocniczej abstrachując wiersz i kolor etc.